src/subshell/common.c

```diff
@@ -618,7 +618,7 @@ read_command_line_buffer (gboolean test_mode)
     FD_ZERO (&read_set);
     FD_SET (command_buffer_pipe[READ], &read_set);
 
-    const int maxfdp = command_buffer_pipe[READ];
+    const int maxfdp = MAX (command_buffer_pipe[READ], mc_global.tty.subshell_pty);
```

Comment:
while not strictly wrong, it seems weird to do this here, as the first loop doesn't actually add the pty fd to the set.